src/apps/Permissions/CreatePermissionPanel.js

```diff
@@ -0,0 +1,331 @@
+import React from 'react'
+import { SidePanel, DropDown, Info, Field, Button, TextInput } from '@aragon/ui'
+import uniqby from 'lodash.uniqby'
+import { PermissionsConsumer } from '../../contexts/PermissionsContext'
+import { isAddress } from '../../web3-utils'
+
+const DEFAULT_STATE = {
+  assignEntityIndex: 0,
+  assignAddressValue: '',
+  appIndex: 0,
+  roleIndex: 0,
+  newRoleValue: '',
+  newRoleManagerValue: '',
+}
+
+// The permission panel, wrapped in a PermissionsContext (see end of file)
+class PermissionPanel extends React.PureComponent {
+  state = {
+    ...DEFAULT_STATE,
+  }
+
+  handleAssignEntityChange = index => {
+    this.setState({ assignEntityIndex: index, assignAddressValue: '' })
+  }
+
+  handleAssignAddressChange = event => {
+    this.setState({ assignAddressValue: event.target.value })
+  }
+
+  handleAppChange = index => {
+    this.setState({ appIndex: index, roleIndex: 0 })
+  }
+
+  handleRoleChange = index => {
+    this.setState({
+      roleIndex: index,
+      newRoleValue: '',
+      newRoleManagerValue: '',
+    })
+  }
+
+  handleNewRoleChange = event => {
+    this.setState({ newRoleValue: event.target.value })
+  }
+
+  handleNewRoleManagerChange = event => {
+    this.setState({ newRoleManagerValue: event.target.value })
+  }
+
+  getNamedApps() {
+    const { apps } = this.props
+    return apps.filter(app => Boolean(app.name))
+  }
+
+  getRoles() {
+    const { getAppRoles } = this.props
+    const app = this.getSelectedApp()
+    const appRoles = app ? getAppRoles(app) : []
+    return uniqby(
+      appRoles
+        .filter(roleData =>
+          Boolean(roleData && roleData.role && roleData.role.bytes)
+        )
+        .map(({ role }) => role),
+      role => role.bytes
+    )
+  }
+
+  appsLabels() {
+    return this.getNamedApps().map(app => app.name)
+  }
+
+  getAssignEntityItems() {
+    return ['Select an entity', ...this.appsLabels(), 'Custom address…']
+  }
+
+  getAppsItems() {
+    return ['Select an app', ...this.appsLabels()]
+  }
+
+  getRolesItems() {
+    const roles = this.getRoles()
+    const names = roles.map(role => role.name || '?')
+    return ['Select a role', ...names, 'Create a new role (advanced)…']
```

Comment:
Unfortunately, we can't create roles just like that haha.

Reply:
It was more meant to set a role that is not assigned to anything => https://github.com/aragon/aragon/pull/334#discussion_r214717396